### DISCUSSION TITLE
Dependency version conflict，The packaged JAR should have only one of these versions？

### DISCUSSION THREAD

**Discussion by lihaoran93** [DependencyTrack/dependency-track#960] [General]
Whether need to exclude one of the versions？
<img width="1545" alt="图片" src="https://user-images.githubusercontent.com/26021760/109937744-2e029c00-7d0a-11eb-98b4-d55c3983ea7b.png">

**Comment by stevespringett** [comment]
It appears the BOM actually includes both versions. Check your BOM to verify. Then check the artifacts created from the build to verify whether or not both versions are included or not. Maven sometimes does this.

**Comment by msymons** [comment]
I have come across so many instances where a BOM includes multiple versions of the same dependency. I find that the `analyze` command in [cyclonedx-cli](https://github.com/CycloneDX/cyclonedx-cli) is very useful here as it explicitly identifies the clashes.  The same information will be displayed in Dependency-Track..  but can be a bit harder to spot in a larger project.

```
$ ./cyclonedx-win-x86.exe analyze --input-file my-bom.xml --output-format text --multiple-component-versions
Components with multiple versions:

javax.ws.rs-api versions: 2.0.1 2.1.1
```

A small tweak to `dependencyManagement` and the next build gives a "clean" BOM.

**Reply by lihaoran93** [reply]
Thanks， Perhaps I have the wrong starting point, as I have been trying to reduce the false positive rate of detected risks:

eg. Unused dependencies in the POM do not want to detect risk.
